### PR TITLE
Extract model_family from model_path automatically by default.

### LINKED ISSUE #1060: Users incorrectly think that by specifying the `--model-family` flag then the correct model is selected for `ilab generate`
Users think by specifying the `--model-family` flag for the `ilab generate` command that it will use the correct model. They are unaware that you also need to specify the model using the `--model` flag. Otherwise, the default model is selected for them and this is unexpected behavior.

The use of the `--model-family` flag can be confusing. Suggest removing the flag in favor of adding the logic to select the correct template in the code for the model of a particular family. A user should not need to know what family a model is from.

From @laurachiticariu:
"
`ilab generate --model-family mixtral`  still seem to use `merlinite` to generate data. Are there instructions on how to make it use `mixtral` instead? (e.g., perhaps I need to serve `mixtral` using some combo of `ilab download` and `ilab serve`
"

This was also raised by @anik120 in https://github.com/instructlab/instructlab/pull/1002#discussion_r1583318874.

### PR DESCRIPTION
**Pull request by makelinux**
Extract model_family from model_path automatically by default.

  --model-family forces model_family.

Resolves #1060

### CODE REVIEW

**Comment by cdoern** [comment]
I am all for implicitly figuring out the model family, but removing the flag isn't necessary.

I vote for 
1. implicitly deciding if the flag isn't specified and 
2. overriding if the user gives it.

Pulling the wool over the user's eyes IMO never goes well, especially with abstracted things like this.

**Comment by lhawthorn** [comment]
I agree with @cdoern's assessment - it is always better to give more information rather than less.

**Comment by makelinux** [comment]
> I agree with @cdoern's assessment - it is always better to give more information rather than less.

I've already implemented it.

**Comment by nathan-weinberg** [comment]
@cdoern is the current implementation good with you?

File: src/instructlab/config.py [hunk 1]
```diff
@@ -190,3 +192,7 @@ def write_config(cfg, config_file=DEFAULT_CONFIG):
 def get_api_base(host_port):
     """Returns server API URL, based on the provided host_port"""
     return f"http://{host_port}/v1"
+
+
+def get_model_family(forced, model_path):
+    return forced if forced else match(r"^\w*", path.basename(model_path)).group(0)
```
Comment (russellb) [inline src/instructlab/config.py:198]: It looks like `match()` can return `None`, but that isn't handled here

**Review by russellb — CHANGES_REQUESTED** [review]
one minor request for change

File: src/instructlab/config.py (hunk 1 above)
Comment (makelinux) [reply src/instructlab/config.py:198]: How can it return `None`?

Comment (nathan-weinberg) [reply src/instructlab/config.py:198]: https://docs.python.org/3/library/re.html#re.match

Comment (makelinux) [reply src/instructlab/config.py:198]: `^\w*` will return at least an empty string. In will not return `None`.

Comment (russellb) [reply src/instructlab/config.py:198]: Ah, right. I didn't read the regex itself closely enough. Thanks.

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**